src/rust/src/lazy/dataframe.rs

```diff
@@ -80,11 +77,77 @@ impl LazyFrame {
         })
     }
 
-    pub fn collect_handled(&self) -> crate::rpolarserr::RResult<crate::rdataframe::DataFrame> {
+    pub fn collect_handled(&self) -> RResult<crate::rdataframe::DataFrame> {
         use crate::rpolarserr::WithRctx;
         handle_thread_r_requests(self.clone().0).when("calling $collect() on LazyFrame")
     }
 
+    pub fn sink_parquet(
+        &self,
+        path: Robj,
+        compression_method: Robj,
+        compression_level: Robj,
+        statistics: Robj,
+        row_group_size: Robj,
+        data_pagesize_limit: Robj,
+        maintain_order: Robj,
+    ) -> RResult<()> {
+        use polars::prelude::ParquetCompression::*;
+        let pqcomp = match robj_to!(String, compression_method)?.as_str() {
```

Comment:
before I would make a separate fn called e.g. new_parquet_compression and place it in rdatatype.rs. To have the conversion looks good also as long as only one method needs it.

Reply:
Refactored